src/Microsoft.ML.TimeSeries/SrCnnEntireTransformer.cs

```diff
@@ -0,0 +1,903 @@
+            public string Target;
+
+            [Argument(ArgumentType.Required, HelpText = "The threshold to determine anomaly, score larger than the threshold is considered as anomaly. Must be in [0,1]. Default value is 0.3.",
+                ShortName = "thre", SortOrder = 101)]
+            public double Threshold = 0.3;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Divide the input data into batches to fit srcnn model. Must be -1 or a positive integer no less than 12. Default value is 1024.",
+                ShortName = "batch", SortOrder = 202)]
+            public int BatchSize = 1024;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The detection mode of SrCnnDetectMode enum type.",
+                ShortName = "mode", SortOrder = 303)]
+            public SrCnnDetectMode SrCnnDetectMode = SrCnnDetectMode.AnomalyOnly;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The sensitivity of boundaries.",
+                ShortName = "sens", SortOrder = 404)]
+            public double Sensitivity = 99;
+        }
+
+        internal readonly string InputColumnName;
+
+        internal readonly string OutputColumnName;
+
+        internal double Threshold { get; }
+
+        internal int BatchSize { get; }
+
+        internal SrCnnDetectMode SrCnnDetectMode { get; }
+
+        internal double Sensitivity { get; }
+
+        internal int OutputLength { get; }
+
+        internal SrCnnEntireModeler AnomalyModel;
+
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "SRENTRNS",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(SrCnnEntireTransformer).Assembly.FullName);
+        }
+
+        internal SrCnnEntireTransformer(IHostEnvironment env, Options options, IDataView input)
+            :base(Contracts.CheckRef(env, nameof(env)).Register(LoaderSignature), new[] { (options.Target, options.Source) })
+        {
+            InputColumnName = options.Source;
+            OutputColumnName = options.Target;
+
+            Host.CheckUserArg(options.Threshold >= 0 && options.Threshold <= 1, nameof(Options.Threshold), "Must be in [0,1].");
+            Threshold = options.Threshold;
+
+            Host.CheckUserArg(options.BatchSize == -1 || options.BatchSize >= 12, nameof(Options.BatchSize), "BatchSize must be -1 or no less than 12.");
+            BatchSize = options.BatchSize;
+
+            SrCnnDetectMode = options.SrCnnDetectMode;
+            if (SrCnnDetectMode.Equals(SrCnnDetectMode.AnomalyOnly))
+            {
+                Sensitivity = options.Sensitivity;
+                OutputLength = 3;
+            }
```

Comment:
Can we please create names for these constants of 3, 4, 7 and use those throughout the code?. E.g. AnomalyOnlyVectorLength, AnomalyAndEVVectorLength, and AnomalyAndMarginVectorLength?
(Or something shorter if possible)

Reply:
> Can we please create names for these constants of 3, 4, 7 and use those throughout the code?. E.g. AnomalyOnlyVectorLength, AnomalyAndEVVectorLength, and AnomalyAndMarginVectorLength?
> (Or something shorter if possible)

Updated as you suggested.